proxy/api/src/main.rs

```diff
@@ -83,15 +83,51 @@ async fn main() -> Result<(), Box<dyn std::error::Error>> {
     let bin_dir = config::bin_dir()?;
     coco::git_helper::setup(&proxy_path, &bin_dir).expect("Git remote helper setup failed");
 
-    log::info!("Starting API");
     let ctx = context::Ctx::from(context::Context {
         peer_api,
         keystore,
         store,
     });
+
+    {
+        let ctx = ctx.clone();
+        log::info!("Starting Announcement watcher");
+        tokio::task::spawn(async move { announcement_watcher(ctx).await });
+    }
+
+    log::info!("Starting API");
     let api = http::api(ctx, args.test);
 
     warp::serve(api).run(([127, 0, 0, 1], 8080)).await;
 
     Ok(())
 }
+
+async fn announcement_watcher(ctx: context::Ctx) {
+    // TODO(xla): Take interval timings from config/settings.
+    let mut timer = tokio::time::interval(std::time::Duration::from_secs(10));
+
+    loop {
+        timer.tick().await;
+
+        let ctx = ctx.read().await;
+
+        let old = announce::load(&ctx.store).expect("unable to load cached announcemnts");
+        let new = announce::build(&ctx.peer_api).expect("unable to build state");
+
+        {
+            let updates = announce::diff(&old, &new);
+
+            ctx.peer_api
+                .with_protocol(|protocol| {
+                    Box::pin(async move { announce::announce(protocol, updates).await })
+                })
+                .await
+                .expect("announce failed");
+        }
+
+        let updates = announce::diff(&old, &new);
+        announce::save(&ctx.store, updates.clone()).expect("unable to save updated announcements");
```

Comment:
We shouldn't forget to `save` when changes are announced during the lifecycle of the application, otherwise these will be-reannounced on startup!

Reply:
Nevermind, this is not only run at startup!